code-studio/src/js/components/pairing/pairing.jsx

```diff
@@ -0,0 +1,169 @@
+/* global $ */
+
+import React from 'react';
+
+import SectionSelector from './section_selector.jsx';
+import StudentSelector from './student_selector.jsx';
+
+/**
+ * A component for managing pair programming.
+ */
+const Pairing = React.createClass({
+  propTypes: {
+    source: React.PropTypes.string,
+    handleClose: React.PropTypes.func
+  },
+
+  getInitialState() {
+    return {
+      pairings: [],
+      sections: []
+    };
+  },
+
+  componentDidMount() {
+    $.ajax({
+      url: this.props.source,
+      method: 'GET',
+      dataType: 'json'
+    }).done(function (result) {
+      this.setState({
+        pairings: result.pairings,
+        sections: result.sections
+      });
+    }.bind(this)).fail(function (result) {
+      // TODO what to do here?
```

Comment:
I think the ideal UX would be to show a spinner in the modal until pairing succeeds, instead of closing immediately.  Then we have a place to show the error message if pairing fails.  I'll try this out.